apteryxd.c

```diff
@@ -479,15 +479,24 @@ call_refreshers (const char *path, bool dry_run)
 
         pthread_mutex_lock (&refresher->lock);
 
-        /* Check if it is time to refresh */
-        if (now < (refresher->timestamp + refresher->timeout))
+        /* We can skip this refresher if the refresher has been called recently AND
+         * the last call was for a path equal to or less specific than this one */
+        if (now < (refresher->timestamp + refresher->timeout) &&
+            strncmp (refresher->last_path, path, strlen (refresher->last_path) - 1) == 0)
```

Comment:
This will fail to refresh /test/animal/name-common if /test/animal/name has recently been updated. It should probably do some strtok stuff to get the answer right.